docs/19-czym-sa-customcommands.md

```diff
@@ -0,0 +1,21 @@
+# Czym są CustomCommands
+
+Aktualnie to jak wyglądają komendy definiują modele, a dokładniej to jakie mają propertiesy i jakich typów są te propertiesy (przykładowo typ numeryczny, tekstowy, czasowy).  
+Przykładowo jeśli komenda zawiera `public int Test { get; set; }` - będzie wymagała argumentu `-test jakaśLiczba`.
+
+Jednak co w przypadku kiedy właściciele/moderatorzy pewnego serwera by zdecydowali, że chcą aby ich komendy nazywały się inaczej?  
+Przykładowo chcieliby zastąpić `-addrole -roles test` komendą `-add role test`, albo idąc dalej `-daj dla mnie role test`.  
+Domyślnym sposobem się nie da - na szczęście przychodzą nam z pomocą komendy niestandardowe, lub tak jak my to nazywamy w systemie - CustomCommands.
+
+## Jak działają CustomCommands
+
+Idea jest bardzo prosta - pomysłodawca wyglądu komendy pisze regexa, który ma grupy.  
+Grupy nazywają się tak jak propertiesy w modelu.  
+A wartości które do nich wpadają, są prawidłowego typu.
```

Comment:
coś mi to zdanie nie do końca pasuje, bo w następnym zdaniu piszesz "czy na pewno wszystkie wartości są prawidłowego typu." a tutaj jakby to już stwierdzasz

Reply:
ja to widze tak że jest
1 założenie jest takie, że powinno być prawidłowo, jak nie będzie to nie zadziała
2 ale dla pewności mamy mechanizm który to wszystko weryfikuje 

ale moge to zmienić, jeśli masz jakąś propozycje

Reply:
może:
A wartości które do nich wpadają, powinny być prawidłowego typu (co później jest walidowane).